NitroxServer/Exceptions/IllegalArgumentException.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Runtime.Serialization;
+
+namespace NitroxServer.Exceptions
+{
+
+    [Serializable]
+    internal class IllegalArgumentException : Exception
```

Comment:
We use ArgumentException and ArgumentNullException throughout.  You can use Validate class as a shortcut to these as well.